src/components/course/data/utils.jsx

```diff
@@ -407,10 +409,344 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+  );
+  const hasExhaustedSubscriptions = subscriptionsApplicableToCourse.every(
+    subscription => subscription?.licenses?.unassigned === 0,
+  );
+  const applicableSubscriptionNonExpiredNonExhausted = subscriptionsApplicableToCourse.find(
+    subscription => subscription.daysUntilExpirationIncludingRenewals >= 0 && subscription?.licenses?.unassigned > 0,
+  );
+
+  if (hasExpiredSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed have expired,
+    // change `reasonType` to use the `SUBSCRIPTION_EXPIRED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (hasExhaustedSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed no longer have
+    // any remaining seats, change `reasonType` to use the `SUBSCRIPTION_SEATS_EXHAUSTED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (subscriptionLicense?.status === LICENSE_STATUS.REVOKED) {
+    // If learner's subscription license is revoked, change `reasonType` to use
+    // the `SUBSCRIPTION_DEACTIVATED` message.
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_DEACTIVATED;
+  }
+
+  if (applicableSubscriptionNonExpiredNonExhausted) {
+    // If customer has a subscription plan(s) containing the course being viewed that is not expired
+    // nor exhausted, change `reasonType` to use the `SUBSCRIPTION_LICENSE_NOT_ASSIGNED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED_NO_ADMINS,
+      },
+    });
+  }
+
+  // There is no applicable subscriptions-related reason for disabled enrollment.
+  return undefined;
+};
+
+export const getEnterpriseOffersDisabledEnrollmentReasonType = ({
+  enterpriseOffers,
+  catalogsWithCourse,
+}) => {
+  if (!enterpriseOffers || enterpriseOffers.length === 0) {
+    return DISABLED_ENROLL_REASON_TYPES.NO_ENTERPRISE_OFFERS;
+  }
+
+  const offersForCourse = enterpriseOffers.filter(offer => catalogsWithCourse.includes(offer.enterpriseCatalogUuid));
+  if (!offersForCourse.length) {
+    return DISABLED_ENROLL_REASON_TYPES.COURSE_NOT_IN_ENTERPRISE_OFFERS;
+  }
+
+  const hasExpiredOffers = offersForCourse.every(offer => !offer.isCurrent);
+  if (hasExpiredOffers) {
+    return DISABLED_ENROLL_REASON_TYPES.ENTERPRISE_OFFERS_EXPIRED;
```

Comment:
There is no `ENTERPRISE_OFFERS_EXPIRED` under `DISABLED_ENROLL_REASON_TYPES`. Is this supposed to be `NO_SUBSIDY`?

Similarly, the other functions that determine whether there's a disabled enroll reason for a particular subsidy type (e.g., subscriptions), all make use of the `parseReasonTypeBasedOnEnterpriseAdmins` helper function to conditionally return the appropriate reason type based on whether the enterprise customer has enterprise administrators or not. The displayed `userMessage` (based on the parsed `reasonType`) is different depending on whether there are enterprise administrators or not.